docs/content/config/advanced/auth-ldap.md

```diff
@@ -6,37 +6,188 @@ title: 'Advanced | LDAP Authentication'
+- DOVECOT_USER_ATTRS=<YOUR_USER_HOME_DIRECTORY_ATTRIBUTE>=home,<YOUR_USER_MAILSTORE_ATTRIBUTE>=mail,<YOUR_USER_MAIL_UID_ATTRIBUTE>=uid,<YOUR_USER_MAIL_GID_ATTRIBUTE>=gid
+```
+
+!!! note
+
+    For `DOVECOT_*_ATTRS`, you can replace `ldapAttr=dovecotAttr` with `=dovecotAttr=%{ldap:ldapAttr}` for more flexibility, like for example `=home=/var/mail/%{ldap:uid}` or just `=uid=5000`.
+
+    A list of dovecot attributes can be found [in the dovecot documentation](https://doc.dovecot.org/configuration_manual/authentication/user_databases_userdb/#authentication-user-database).
+
+???+ example "Example Code: Setup for a directory that has the [qmail-schema](https://github.com/amery/qmail/blob/master/qmail.schema) installed and that uses `uid`"
+
+    ```yaml
+    - DOVECOT_PASS_ATTRS=uid=user,userPassword=password
+    - DOVECOT_USER_ATTRS=homeDirectory=home,qmailUID=uid,qmailGID=gid,mailMessageStore=mail
+    - DOVECOT_PASS_FILTER=(&(objectClass=qmailUser)(uid=%u)(accountStatus=active))
+    - DOVECOT_USER_FILTER=(&(objectClass=qmailUser)(uid=%u)(accountStatus=active))
+    ```
+
+The LDAP server configuration for dovecot will be taken mostly from postfix, other options can be found in [`ENVIRONMENT.md`][github-file-env].
+
+### `SASLAUTHD_LDAP_FILTER`
+This filter is used for `saslauthd`, which is called by postfix when someone is authenticating through SMTP (assuming that `SASLAUTHD_MECHANISMS=ldap` is being used). Note that you'll need to set up the LDAP server for saslauthd seperately from postfix.
+
+The filter variables are explained in detail [in the `LDAP_SASLAUTHD` file](https://github.com/winlibs/cyrus-sasl/blob/master/saslauthd/LDAP_SASLAUTHD#L121), but unfortunately, this method doesn't really support domains right now - that means that `%U` is the only token that makes sense in this variable.
+
+!!! note "When to use this and how to avoid it"
+
+    Using a separate filter for SMTP authentication allows you to for example allow `noreply@example.org` to send email, but not log in to IMAP or receive email: `(&(mail=%U@example.org)(|(memberOf=cn=email,*)(mail=noreply@example.org)))`
+
+    If you don't want to use a separate filter for SMTP authentication, you can set `SASLAUTHD_MECHANISMS=rimap` and `SASLAUTHD_MECH_OPTIONS=127.0.0.1` to authenticate against dovecot instead - this means that the `DOVECOT_USER_FILTER` and `DOVECOT_PASS_FILTER` will be used for SMTP authentication as well.
+
+???+ example "Example Code: LDAP with saslauthd"
+
+    ```yaml
+    - ENABLE_SASLAUTHD: "1"
+    - SASLAUTHD_MECHANISMS: "ldap"
+    - SASLAUTHD_LDAP_SERVER: "ldap.example.org"
+    - SASLAUTHD_LDAP_BIND_DN: "cn=admin,dc=example,dc=org"
+    - SASLAUTHD_LDAP_PASSWORD: "mypassword"
+    - SASLAUTHD_LDAP_SEARCH_BASE: "ou=users,dc=example,dc=org"
+    - SASLAUTHD_LDAP_FILTER: "(mail=%U@example.org)"
+    ```
+
+## Secure Connection with LDAPS or StartTLS
+TODO
+
+## LDAP Setup Examples
+
+### Basic Setup
```

Comment:
If the intention is to just provide some config snippets here with no other content, you could drop the headings and collapse the admonitions (`???` instead of `???+`), then an interested user clicks it to expand the config example.

I don't mind either way.

Reply:
I purposefully expanded the basic setup example, as that's probably what most people are coming for to the documentation. Is that fine or do you think that the example is too long?

Reply:
I'm fine with it for the same reason you stated

Reply:
I agree that we should remove the headers though, or at least make the Kopano header the same level